### PR TITLE
{issuegen,motdgen}.service: `sleep 1` before we start generating

### PR DESCRIPTION
**Pull request by kelvinfan001**
This imposes throttling and ensures we're not constantly regenerating
when things change quickly (as can happen during system startup).
`sleep` in `ExecStartPre` so we capture *all* issue snippets dropped
into `/run` in the one second interval.
Also set rate limit to once per second so we would never hit  any
systemd 'start-limit-hit' failures for our .path and .service units.

xref https://github.com/coreos/console-login-helper-messages/issues/87

### CODE REVIEW

File: usr/lib/systemd/system/console-login-helper-messages-issuegen.service [hunk 1]
```diff
@@ -17,6 +21,11 @@ Type=oneshot
 # effect).
 RemainAfterExit=no
 ExecStart=/usr/libexec/console-login-helper-messages/issuegen
+# See https://github.com/coreos/console-login-helper-messages/issues/87
+# We want to avoid service start limits and throttle the rate at
+# which we regenerate.  This ensures the service starts at most once
+# per second.
+ExecStartPre=/bin/sh -c 'sleep 1'
```
Comment (dustymabe) [inline usr/lib/systemd/system/console-login-helper-messages-issuegen.service:28]:
```suggestion
ExecStartPre=sleep 1
```

Looks a little cleaner. Also WDYT about moving the `ExecStartPre` before the `ExecStart` (it helps when thinking about the unit to see order the entries in the order the operations are going to run in).

**Review by dustymabe — APPROVED** [review]
LGTM. Seems to work in local testing of RHCOS 4.6

File: usr/lib/systemd/system/console-login-helper-messages-issuegen.service (hunk 1 above)
Comment (kelvinfan001) [reply usr/lib/systemd/system/console-login-helper-messages-issuegen.service:28]: Looks a lot cleaner :D 
For my knowledge though, when should we execute scripts like this in a separate process?

Comment (jlebon) [reply usr/lib/systemd/system/console-login-helper-messages-issuegen.service]: Using `sh -c` is useful if e.g. you need shell expansion or want to have some basic conditional logic, etc... But in general if it's just calling out to one program, it's better to call out to that one specifically. That removes one layer from the hierarchy and makes monitoring and logging from the service nicer.

**Comment by kelvinfan001** [comment]
@dustymabe Added some tests; they are passing, but an extra sanity check would be nice :)

**Pull request merged**